src/module_3/Module3_Milestone1.md

```diff
@@ -0,0 +1,1825 @@
+
+
+# Add legends and show plots
+axes_train[0].legend(loc='upper right')
+axes_train[1].legend(loc='lower right')
+axes_val[0].legend(loc='upper right')
+axes_val[1].legend(loc='lower right')
+
+plt.tight_layout()
+fig_train.show()
+fig_val.show()
+```
+
+    /tmp/ipykernel_82778/2610710267.py:50: UserWarning: FigureCanvasAgg is non-interactive, and thus cannot be shown
+      fig_train.show()
+    /tmp/ipykernel_82778/2610710267.py:51: UserWarning: FigureCanvasAgg is non-interactive, and thus cannot be shown
+      fig_val.show()
+
+
+
+    
+![png](Module3_Milestone1_files/Module3_Milestone1_47_1.png)
+    
+
+
+
+    
+![png](Module3_Milestone1_files/Module3_Milestone1_47_2.png)
+    
+
+
+We can see that the model's performance remains consistent with previous outcomes, despite using only a limited set of variables for both models.
+
+### 5. Training a model with categorical variables
+
+#### Addressing categorical variables
+Logistic regression models in libraries like Scikit-learn expect all input features to be numeric. Therefore, categorical variables need to be converted into a numerical format before they can be used for model training. It is worth remembering that we do not consider variant_id, order_id, and user_id categorical variables, but informative ones, and these will not be used to train the model.
+
+First, deal with the two datetime columns. For created_at, I split the information of these into four columns: for year, month, day and hour. On the other hand, since order_date does not contain any additional information, I remove it.
+
+
+```python
+def transform_datetime(df, date_col):
+    """
+    Transforms a datetime column into the following extracted components: year, month, day, and hour.
+    
+    Input parameters:
+    df (pd.DataFrame): The DataFrame containing the datetime column.
+    date_col (datetome): The name of the datetime column to transform.
+    
+    Returns:
+    pd.DataFrame: The modified DataFrame with new columns for date components.
+    """
+
+    # Check if the column exists in the DataFrame
+    if date_col not in df.columns:
+        raise ValueError(f"The column {date_col} does not exist in the DataFrame.")
+    
+    # Extracting components
+    df[f'{date_col}_year'] = df[date_col].dt.year
+    df[f'{date_col}_month'] = df[date_col].dt.month
+    df[f'{date_col}_day'] = df[date_col].dt.day
+    df[f'{date_col}_hour'] = df[date_col].dt.hour
+    
+    # Remove the original datetime column
+    df = df.drop(columns=[date_col])
+    
+        
+    return df
+```
+
+
+```python
+extended_cols = reduced_cols + categorical_cols + ['created_at']
+X_train_cat = X_train[extended_cols]
+X_val_cat = X_val[extended_cols]
+
+# Transform created_at column
+X_train_cat = transform_datetime(X_train_cat, 'created_at')
+X_val_cat = transform_datetime(X_val_cat, 'created_at')
+   
+print(X_train_cat.shape)
+X_train_cat.head()
+```
```

Comment:
I'd recomend to use .loc everywhere to avoid potential problems, unless you have a clear undestanding of the warning below and you are sure it is ok the curren approach